packages/itwin/grouping-mapping-widget/src/components/Mappings/MappingListItem.tsx

```diff
@@ -0,0 +1,132 @@
+import { ExtractionStatus } from "./Extraction/ExtractionStatus";
+import { useExtractionStateJobContext } from "../context/ExtractionStateJobContext";
+import { resetMappingExtractionStatus } from "./hooks/useFetchMappingExtractionStatus";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useFetchMappingExtractionStatus } from "./hooks/useFetchMappingExtractionStatus";
+import "./MappingListItem.scss";
+import { useGroupingMappingApiConfig } from "../context/GroupingApiConfigContext";
+import { useIsMounted } from "../../common/hooks/useIsMounted";
+
+export interface MappingListItemProps {
+  selected: boolean;
+  onSelectionChange: (mapping: Mapping) => void;
+  mapping: Mapping;
+  jobId: string;
+  jobStartEvent: BeEvent<(mappingId: string) => void>;
+  onClickMappingTitle?: (mapping: Mapping) => void;
+  onClickMappingModify?: (mapping: Mapping) => void;
+  onRefreshMappings: () => Promise<void>;
+  onToggleExtraction: (mapping: Mapping) => Promise<void>;
+  setShowDeleteModal: (mapping?: Mapping) => void;
+}
+
+export const MappingListItem = (props: MappingListItemProps) => {
+  const [extractionState, setExtractionState] = useState<ExtractionStates | undefined>(ExtractionStates.None);
+  const { mappingIdJobInfo, setMappingIdJobInfo } = useExtractionStateJobContext();
+  const groupingMappingApiConfig = useGroupingMappingApiConfig();
+  const [isJobStarted, setIsJobStarted] = useState<boolean>(false);
+  const [isListItemMounted, setIsListItemMounted] = useState<boolean>(false);
+  const isMounted = useIsMounted();
+  const { statusQuery } = useFetchMappingExtractionStatus({isMounted: isListItemMounted, ...groupingMappingApiConfig, mapping: props.mapping, enabled: isJobStarted});
+  const queryClient = useQueryClient();
+
+  const onClickTile = () => {
+    props.onSelectionChange(props.mapping);
+  };
+
+  useEffect(() => {
+    if(isMounted()){
+      setIsListItemMounted(true);
+      setIsJobStarted(true);
+    }
+  }, [isMounted]);
+
+  // Check whether the job is still running when users refresh the mapping list
+  // or modify any mappings
+  useEffect(() => {
+    if(mappingIdJobInfo.get(props.mapping.id)){
+      setIsJobStarted(true);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [mappingIdJobInfo]);
+
+  const resolveTerminalExtractionStatus = useCallback(async () => {
+    if(statusQuery.data!.finalExtractionStateValue === ExtractionStates.Failed || statusQuery.data!.finalExtractionStateValue === ExtractionStates.Succeeded){
+      setIsJobStarted(false);
+      setMappingIdJobInfo((prevMap: Map<string, string>) => {
+        const newMap = new Map(prevMap);
+        newMap.delete(props.mapping.id);
+        return newMap;
+      });
+      await resetMappingExtractionStatus(props.mapping.id, queryClient);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [statusQuery, props.mapping.id]);
```

Comment:
Why not pull all required dependencies in here? This callback might become stale.

Reply:
I included all the dependencies.